numpy/typing/tests/data/reveal/ma.pyi

```diff
@@ -623,3 +623,112 @@ assert_type(AR_LIKE_c - MAR_o, Any)
 assert_type(AR_LIKE_td64 - MAR_o, Any)
 assert_type(AR_LIKE_dt64 - MAR_o, Any)
 assert_type(AR_LIKE_o - MAR_o, Any)
+
+# Masked Array multiplication
+
+assert_type(MAR_b * AR_LIKE_u, MaskedArray[np.uint32])
+assert_type(MAR_b * AR_LIKE_i, MaskedArray[np.signedinteger])
+assert_type(MAR_b * AR_LIKE_f, MaskedArray[np.floating])
+assert_type(MAR_b * AR_LIKE_c, MaskedArray[np.complexfloating])
+assert_type(MAR_b * AR_LIKE_td64, MaskedArray[np.timedelta64])
+assert_type(MAR_b * AR_LIKE_o, Any)
+
+# assert_type(AR_LIKE_u * MAR_b, MaskedArray[np.uint32])
+# assert_type(AR_LIKE_i * MAR_b, MaskedArray[np.signedinteger])
+# assert_type(AR_LIKE_f * MAR_b, MaskedArray[np.floating])
+# assert_type(AR_LIKE_c * MAR_b, MaskedArray[np.complexfloating])
+# assert_type(AR_LIKE_td64 * MAR_b, MaskedArray[np.timedelta64])
+# assert_type(AR_LIKE_o * MAR_b, Any)
```

Comment:
I can't get these commented-out ones to pass

The first one, for example, `AR_LIKE_u * MAR_b` is:
- exactly the reverse of `MAR_b * AR_LIKE_u`, which passes
- `MaskedArray.__mul__` has the following overload:
  ```python
  @overload
  def __mul__(self: _MaskedArray[np.bool], other: _ArrayLike[_NumberT], /) -> _MaskedArray[_NumberT]: ...  # type: ignore[overload-overlap]
  ```
  while `MaskedArray.__rmul__` has
  ```python
  @overload
  def __rmul__(self: _MaskedArray[np.bool], other: _ArrayLike[_NumberT], /) -> _MaskedArray[_NumberT]: ...  # type: ignore[overload-overlap]
  ```

Reply:
What does mypy say the type is, instead?